hydradx/other tests/test_price_manipulations.py

```diff
@@ -0,0 +1,479 @@
+        )
+
+        # close arb
+        glob = lp_agent.trade_strategy.execute(
+            state=GlobalState(
+                agents={
+                    'agent': lp_agent.copy(),
+                },
+                pools={
+                    'omnipool': lp_state.copy(),
+                },
+                external_market=market_prices
+            ), agent_id='agent'
+        )
+
+        final_state: oamm.OmnipoolState = glob.pools['omnipool']
+        final_agent: Agent = glob.agents['agent']
+
+        final_diff = {
+            tkn: final_agent.holdings[tkn] - (initial_agent.holdings[tkn] if tkn in initial_agent.holdings else 0) for
+            tkn in final_agent.holdings}
+
+        profit = (
+                oamm.cash_out_omnipool(final_state, final_agent, market_prices)
+                - oamm.cash_out_omnipool(initial_state, initial_agent, market_prices)
+        )
+
+        if profit > max_profit[0]:
+            max_profit = (profit, buy_quantity, lp_quantity_1, lp_quantity_2)
+
+    print()
+    print(f'# agent buys {max_profit[1]} {asset2} with {asset1}')
+    print(f'# agent LPs {max_profit[2]} {asset1}.')
+    print(f'# agent LPs {max_profit[3]} {asset2}.')
+    print(f'# agent arbitrages all assets and withdraws all liquidity.')
+    print(f'# agent nets {max_profit[0]}')
+
+
+@given(
+    st.floats(min_value=100000.0, max_value=10000000.0),
+    st.floats(min_value=100000.0, max_value=10000000.0)
+)
+def test_price_manipulation(usd_liquidity, dai_liquidity):
+    omnipool: oamm.OmnipoolState = oamm.OmnipoolState(
+        tokens={
+            'HDX': {'liquidity': 1000000, 'LRNA': 1000000},
+            'WETH': {'liquidity': 2265161, 'LRNA': 2265161},
+            'DAI': {'liquidity': 2254499, 'LRNA': 2254499},
+        },
+        lrna_fee=0,
+        asset_fee=0,
+        preferred_stablecoin='DAI',
+        # remove_liquidity_volatility_threshold=0  # test will fail if this is uncommented
+    )
+
+    initial_agent = Agent(
+        holdings={tkn: omnipool.liquidity[tkn] / 2 for tkn in ['WETH', 'DAI']},
+        trade_strategy=price_manipulation(
+            pool_id='omnipool',
+            asset1='WETH',
+            asset2='DAI'
+        )
+    )
+    initial_agent.holdings['LRNA'] = 0
+
+    state = GlobalState(
+        pools={'omnipool': omnipool},
+        agents={'agent': initial_agent},
+        external_market={tkn: oamm.usd_price(omnipool, tkn) for tkn in omnipool.asset_list}
+    )
+
+    start_holdings = copy.deepcopy(initial_agent.holdings)
+    events = run.run(state, 10, silent=True)
+
+    profit = (
+        oamm.cash_out_omnipool(events[-1].pools['omnipool'], events[-1].agents['agent'], state.external_market)
+        - oamm.cash_out_omnipool(omnipool, initial_agent, state.external_market)
+    )
+
+    holdings = events[-1].agents['agent'].holdings
+    er = 1
+    if profit > 0:
+        raise AssertionError(f'Profit: {profit}, Holdings: {holdings}')
+
+
+@given(
```

Comment:
I'm confused why one of these "fuzz" tests uses the hypothesis functionality and one does not?